.github/workflows/community_ci.yml

```diff
@@ -62,68 +30,148 @@ jobs:
+    runs-on: tooling-test-small
+    timeout-minutes: 20
+    permissions:
+      statuses: write
+    env:
+      GITHUB_HEAD_REF: ${{ github.head_ref }}
+      MAIN_BRANCH_NAME: "augustin/04-18-community-ci_workflow_prevent_injection"
+      # TODO
+      #MAIN_BRANCH_NAME: "master"
+
     steps:
-      - name: Checkout Airbyte
-        uses: actions/checkout@v3
+      - name: Checkout fork
+        uses: actions/checkout@v4
         with:
-          # This can checkouts forks
-          # /!\ untrusted code
+          # This checkouts a fork which can contain untrusted code
           # It's deemed safe as the community-ci environment requires manual reviewer approval to run
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
           ref: ${{ github.head_ref }}
           fetch-depth: 1
-      - name: Extract branch name [WORKFLOW DISPATCH]
-        shell: bash
-        if: github.event_name == 'workflow_dispatch'
-        run: echo "branch=${GITHUB_REF#refs/heads/}" >> $GITHUB_OUTPUT
-        id: extract_branch
-      - name: Fetch last commit id from remote branch [PULL REQUESTS]
-        if: github.event_name == 'pull_request_target'
+
+      # On checkout the head commit is a detached head
+      # We have to use the git CLI to get the last commit id of the head branch
+      # We also perform validation and sanitization on the GITHUB_HEAD_REF to prevent command injection
+      - name: Fetch last commit id from remote branch
         id: fetch_last_commit_id_pr
-        run: echo "commit_id=$(git ls-remote --heads origin ${{ github.head_ref }} | cut -f 1)" >> $GITHUB_OUTPUT
-      - name: Fetch last commit id from remote branch [WORKFLOW DISPATCH]
-        if: github.event_name == 'workflow_dispatch'
-        id: fetch_last_commit_id_wd
-        run: echo "commit_id=$(git rev-parse origin/${{ steps.extract_branch.outputs.branch }})" >> $GITHUB_OUTPUT
-      - name: Test connectors [WORKFLOW DISPATCH]
-        if: github.event_name == 'workflow_dispatch'
+        run: |
+
+          # Validation: Check if GITHUB_HEAD_REF contains only alphanumeric characters, hyphens, underscores or slashes
+          if [[ ! "$GITHUB_HEAD_REF" =~ ^[a-zA-Z0-9_/-]+$ ]]; then
+            echo "Invalid input: '${GITHUB_HEAD_REF}'. Only alphanumeric characters, hyphens, underscores or slashes are allowed."
+            exit 1
+          fi
+
+          # Sanitization: Remove any characters that are not alphanumeric, hyphens, underscores, or slashes
+          sanitized_head_ref=$(echo "$GITHUB_HEAD_REF" | tr -cd '[:alnum:]-_/')
+          echo "commit_id=$(git ls-remote --heads origin '${sanitized_head_ref}' | cut -f 1)" >> $GITHUB_OUTPUT
+          echo $sanitized_head_ref
+
+      # This will sync the .github folder of the main repo with the fork
+      # This allows us to use up to date actions from the main repo
+      - name: Pull .github folder from main repository
```

Comment:
This enforces that actions are always from our master branch, and not changed in the fork? Why not just guard the whole run with a check that CI itself isn't changeD?

Reply:
Because I think we have no guarantee that the fork is up to date with  main repo's `master` and has the most up to date version of our CI logic in `action`.

I faced this problem during my iteration: I opened a fork, made changes the actions on the main repo, and did not understand why the logic of the local action in `.github` was not updated.

Reply:
Ah, so we want to ensure that they ran new versions of tests and such?